backend/server.go

```diff
@@ -8,29 +8,97 @@ import (
 	"github.com/go-chi/chi/v5"
 	"github.com/go-chi/chi/v5/middleware"
 	"github.com/go-chi/cors"
+	"golang.org/x/crypto/acme/autocert"
 	"net/http"
+	"os"
+	"path/filepath"
 	"strings"
 )
 
 // See https://github.com/go-chi/chi for documentation
 
-const PORT = 5001 // todo: put into environment var
+func init() {
+
+	if value, ok := os.LookupEnv("NODE_ENV"); ok && value == "production" {
+		allVars := []envVar{
+			{key: "API_BASE_URL", preset: "/api"},
+		}
+
+		err := filepath.Walk("/static/_next", func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			if !info.IsDir() && filepath.Ext(path) == ".js" {
+				err := replaceVarsInFile(path, allVars)
+				if err != nil {
+					fmt.Printf("Error replacing vars in file %s: %v\n", path, err)
+				}
+			}
+
+			return nil
+		})
+
+		if err != nil {
+			fmt.Printf("Error walking the path: %v\n", err)
+		}
+	}
+}
 
 func main() {
+	var origins []string
+	if value, ok := os.LookupEnv("NODE_ENV"); ok && value == "production" {
+		if domain, ok := os.LookupEnv("WEBAUTHN_DOMAIN"); ok {
```

Comment:
maybe we can call this `APP_DOMAIN` or `CLIENT_DOMAIN` something, since its going to cors?